tests/tck/openCypher/features/expressions/list/List2.feature

```diff
@@ -0,0 +1,183 @@
+
+  Scenario: [1] List slice
+    Given any graph
+    When executing query:
+      """
+      WITH [1, 2, 3, 4, 5] AS list
+      RETURN list[1..3] AS r
+      """
+    Then the result should be, in any order:
+      | r      |
+      | [2, 3] |
+    And no side effects
+
+  Scenario: [2] List slice with implicit end
+    Given any graph
+    When executing query:
+      """
+      WITH [1, 2, 3] AS list
+      RETURN list[1..] AS r
+      """
+    Then the result should be, in any order:
+      | r      |
+      | [2, 3] |
+    And no side effects
+
+  Scenario: [3] List slice with implicit start
+    Given any graph
+    When executing query:
+      """
+      WITH [1, 2, 3] AS list
+      RETURN list[..2] AS r
+      """
+    Then the result should be, in any order:
+      | r      |
+      | [1, 2] |
+    And no side effects
+
+  Scenario: [4] List slice with singleton range
+    Given any graph
+    When executing query:
+      """
+      WITH [1, 2, 3] AS list
+      RETURN list[0..1] AS r
+      """
+    Then the result should be, in any order:
+      | r   |
+      | [1] |
+    And no side effects
+
+  Scenario: [5] List slice with empty range
+    Given any graph
+    When executing query:
+      """
+      WITH [1, 2, 3] AS list
+      RETURN list[0..0] AS r
+      """
+    Then the result should be, in any order:
+      | r  |
+      | [] |
+    And no side effects
+
+  Scenario: [6] List slice with negative range
+    Given any graph
+    When executing query:
+      """
+      WITH [1, 2, 3] AS list
+      RETURN list[-3..-1] AS r
+      """
+    Then the result should be, in any order:
+      | r      |
+      | [1, 2] |
+    And no side effects
+
+  Scenario: [7] List slice with invalid range
+    Given any graph
+    When executing query:
+      """
+      WITH [1, 2, 3] AS list
+      RETURN list[3..1] AS r
+      """
+    Then the result should be, in any order:
+      | r  |
+      | [] |
+    And no side effects
+
+  Scenario: [8] List slice with exceeding range
+    Given any graph
+    When executing query:
+      """
+      WITH [1, 2, 3] AS list
+      RETURN list[-5..5] AS r
+      """
+    Then the result should be, in any order:
+      | r         |
+      | [1, 2, 3] |
+    And no side effects
+
+  Scenario Outline: [9] List slice with null range
+    Given any graph
+    When executing query:
+      """
+      WITH [1, 2, 3] AS list
+      RETURN list[<lower>..<upper>] AS r
+      """
+    Then the result should be, in any order:
+      | r    |
+      | null |
+    And no side effects
+
+    Examples:
+      | lower | upper |
+      | null  | null  |
+      | 1     | null  |
+      | null  | 3     |
+      |       | null  |
+      | null  |       |
+
+  @skip
+  Scenario: [10] List slice with parameterised range
+    Given any graph
+    And parameters are:
+      | from | 1 |
+      | to   | 3 |
+    When executing query:
+      """
+      WITH [1, 2, 3] AS list
+      RETURN list[$from..$to] AS r
```

Comment:
$var = YIELD 3 as from, 5 as to;
YIELD [1, 2, 3][$var.from..$var.to] AS r;

Reply:
Had add.